perf(linter/no-unwanted-polyfillio): should run when source type is jsx

shulaoda opened the pull request:
I believe some of the rules under Next.js should only run when the source type variant is `jsx`. If that’s correct, I’ll gradually refine those rules across the next few PRs.

shulaoda commented:
* **#15204** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/15204?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/15204?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=15204).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented:
### Merge activity

* **Nov 1, 12:11 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 1, 12:11 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).
* **Nov 1, 12:16 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

shulaoda committed:
perf(linter/no-unwanted-polyfillio): should run when source type is jsx (#15204)

I believe some of the rules under Next.js should only run when the source type variant is `jsx`. If that’s correct, I’ll gradually refine those rules across the next few PRs.

shulaoda commented on crates/oxc_linter/src/rules/nextjs/no_unwanted_polyfillio.rs:
```diff
@@ -48,6 +48,10 @@ declare_oxc_lint!(
+    fn should_run(&self, ctx: &ContextHost) -> bool {
+        ctx.source_type().is_jsx()
+    }
+
```
Yes. I think you are right.
https://github.com/oxc-project/oxc/blob/af261a00ff2f054328f71af6f50277c35abda27c/crates/oxc_linter/src/lib.rs#L184-L191
